libethcore/Transaction.cpp

```diff
@@ -163,12 +163,12 @@ void TransactionBase::checkChainId(int chainId) const
 		BOOST_THROW_EXCEPTION(InvalidSignature());
 }
 
-bigint TransactionBase::gasRequired(bool _contractCreation, bytesConstRef _data, EVMSchedule const& _es, u256 const& _gas)
+int64_t TransactionBase::baseGasRequired(bool _contractCreation, bytesConstRef _data, EVMSchedule const& _es)
 {
-	bigint ret = (_contractCreation ? _es.txCreateGas : _es.txGas) + _gas;
+	int64_t g = _contractCreation ? _es.txCreateGas : _es.txGas;
 	for (auto i: _data)
```

Comment:
The 63-bit precision is big enough. The cost is 68 gas units per byte, so you will need 115 PB of input to overflow.

I also dream about using Undefined Behavior Sanitizer (aka UBSan) at some point. It will detect signed integer overflows at runtime. That's also a reason for using signed integers for gas calculations.